pkg/generate/app/app.go

```diff
@@ -109,32 +111,46 @@ func (r *SourceRef) SuggestName() (string, bool) {
 	if len(r.Name) > 0 {
 		return r.Name, true
 	}
-	return nameFromGitURL(r.URL)
+	if r.URL != nil {
+		return nameFromGitURL(r.URL)
+	}
+	return "", false
 }
 
 // BuildSource returns an OpenShift BuildSource from the SourceRef
 func (r *SourceRef) BuildSource() (*buildapi.BuildSource, []buildapi.BuildTriggerPolicy) {
-	return &buildapi.BuildSource{
+	triggers := []buildapi.BuildTriggerPolicy{
+		{
+			Type: buildapi.GitHubWebHookBuildTriggerType,
+			GitHubWebHook: &buildapi.WebHookTrigger{
+				Secret: generateSecret(20),
+			},
+		},
+		{
+			Type: buildapi.GenericWebHookBuildTriggerType,
+			GenericWebHook: &buildapi.WebHookTrigger{
+				Secret: generateSecret(20),
+			},
+		},
```

Comment:
> Good point. Maybe you don't want it? But we should probably use the defaulter on a lot of other stuff. We could probably reuse the secret too.

you might not..but the distinction right now is "if you do a source build, you want it, and if you do a docker build you don't", that seems pretty arbitrary as it is, might as well be consistent in shoving it down people's throats.


(though i'm also not even clear on under which circumstances this method wouldn't be called by BuildConfig().. when do you ever invoke BuildConfig but not have source?  when defining a custom strategy?  does new-app even support that?  or is it for new-build?)